### PR TITLE
Add ability to clear ALL data associated with an index

### PR DESCRIPTION
**Pull request by tylerhutcherson**
This PR introduces the `clear()` method to the core SearchEngine class in RedisVL. This now allows for the ability to clear out all data, while leaving the index in place.

Useful for manual cache eviction, manual session clearing, and more.

This PR also updates the extension classes to use the new `clear()` method as opposed to the SCAN ITER approach.

### CODE REVIEW

File: redisvl/index/index.py
Comment (bsbodden) [inline redisvl/index/index.py:1]: Maybe we should let them control the batching parameter but defaulted to 500?

File: redisvl/index/index.py [hunk 1]
```diff
@@ -476,6 +477,26 @@ def delete(self, drop: bool = True):
...
+
+        Returns:
+            int: Count of records deleted from Redis.
+        """
+        # Track deleted records
+        total_records_deleted: int = 0
+
+        # Paginate using queries and delete in batches
+        for batch in self.paginate(
+            FilterQuery(FilterExpression("*"), return_fields=["id"]), page_size=500
```
Comment (bsbodden) [inline redisvl/index/index.py:492]: Unlikely scenario, but what if I want to get the (some) data out before destroying the index, e.g. `return_fields` being configurable also?

**Review by bsbodden — COMMENTED** [review]
LGTM, left a couple of comments

File: redisvl/index/index.py (hunk 1 above)
Comment (tylerhutcherson) [reply redisvl/index/index.py:492]: I agree with the notion that someone might want to be able to export data from an index. However, I think that should be it's own clear feature. For example, an `index.export(file_path="file.json")` method or something similar?

I think `clear()` should just be a simple method for clearing data out of your index

File: redisvl/index/index.py
Comment (tylerhutcherson) [reply redisvl/index/index.py:1]: I think if someone's goal is to just clear all of the data from an index, we should implement the best practice under the hood and not make the user worry about it? But it certainly woudln't be hard to add an optional arg in the future if folks need it in the future!

**Pull request merged**